frontend/src/Reducers/wishlistReducer.js

```diff
@@ -0,0 +1,47 @@
+const initialWishlistData = {
+  wishlistItems: JSON.parse(localStorage.getItem("maxStoreUserWishlist")) || [],
+};
+
+const wishlistReducer = (state, action) => {
+  switch (action.type) {
+    case "WISHLIST_ADD_ITEM":
+      const item = action.payload;
+
+      const existItem = state.wishlistItems.find((x) => x._id === item._id);
+
+      if (existItem) {
+        return state;
+      } else {
+        const newState = {
+          ...state,
+          wishlistItems: [...state.wishlistItems, item],
+        };
+        localStorage.setItem(
+          "maxStoreUserWishlist",
+          JSON.stringify(newState.wishlistItems)
+        );
+        return newState;
+      }
+    case "WISHLIST_REMOVE_ITEM":
+      const newState = {
+        ...state,
+        wishlistItems: state.wishlistItems.filter(
+          (x) => x._id !== action.payload
+        ),
+      };
+      localStorage.setItem(
```

Comment:
suggestion - as reducers are supposed to be pure functions. updating `localStorage` here doesn't seem right to me. It can be moved to the event/function that is dispatching this action. Please correct me if I'm wrong.

Reply:
Sounds good, thanks for pointing out. I will make the changes in a future PR.